tests/cli/test_base.py

```diff
@@ -21,11 +22,19 @@
 from zenml.core.constants import ZENML_DIR_NAME
 
 
-def test_init_creates_zen_folder(tmp_path):
+def test_init_creates_zen_folder(tmp_path: Path) -> None:
     """Check that init command creates a .zen folder inside temporary directory"""
     runner = CliRunner()
     Repo.init(tmp_path, mkdir=True)
     repository_path = tmp_path
     runner.invoke(init, ["--repo_path", str(repository_path)])
     dir_files = os.listdir(repository_path)
     assert ZENML_DIR_NAME in dir_files
+
+
+def test_init_raises_error_when_repo_not_git_repo(tmp_path: Path) -> None:
+    """Ensure ZenML fails when the given path is not a git repository"""
+    runner = CliRunner()
+    zen_fake_repo_path = tmp_path / ZENML_DIR_NAME
+    result = runner.invoke(init, ["--repo_path", str(zen_fake_repo_path)])
+    assert result.exit_code == 2
```

Comment:
Yeah I wonder if that's a road we should or shouldn't go down. It is possible to test the function independently of the `CLIRunner` / CLI execution, but then essentially we might end up testing everything twice? What do you think?

Reply:
Yeah that might duplicate everything, I guess in the end it doesn't really matter what Exception gets raised in this case. I can't think of any good solution to that unless there is something in the CLIRunner, otherwise I think we can leave it as is.